tik/tikSwap/tikSwap.cpp

```diff
@@ -32,53 +34,40 @@ cl::opt<bool> ASCIIFormat("S", cl::desc("output json as human-readable ASCII tex
 int main(int argc, char *argv[])
 {
     cl::ParseCommandLineOptions(argc, argv);
-    //load the original bitcode
+    spdlog::info("Starting tikSwap.");
+    // load the original bitcode
     LLVMContext OContext;
     SMDiagnostic Osmerror;
     unique_ptr<Module> sourceBitcode;
     try
     {
         sourceBitcode = parseIRFile(OriginalBitcode, Osmerror, OContext);
     }
+    // if the file was not found
     catch (exception &e)
     {
-        std::cerr << "Couldn't open input bitcode file: " << OriginalBitcode << "\n";
-        std::cerr << e.what() << '\n';
-        spdlog::critical("Failed to open source bitcode: " + OriginalBitcode);
+        spdlog::critical("Failed to open source bitcode file: " + OriginalBitcode);
         return EXIT_FAILURE;
     }
+    // if the parsing failed
     if (sourceBitcode == nullptr)
     {
-        std::cerr << "Couldn't open input bitcode file: " << OriginalBitcode << "\n";
-        spdlog::critical("Failed to open source bitcode: " + OriginalBitcode);
+        spdlog::critical("Couldn't parse source bitcode: " + OriginalBitcode);
         return EXIT_FAILURE;
     }
     // Annotate its bitcodes and values
-    Module *base = sourceBitcode.get();
-    CleanModule(base);
-    Format(base);
-    // create a map for its block and value IDs
-    map<int64_t, BasicBlock *> baseBlockMap;
-    for (auto &F : *base)
+    CleanModule(sourceBitcode.get());
+    Format(sourceBitcode.get());
+    // change all non-external function definitions to default linkage type
+    for (auto &F : *sourceBitcode)
     {
-        for (auto BB = F.begin(), BBe = F.end(); BB != BBe; BB++)
+        if (!F.hasExternalLinkage())
         {
-            auto block = cast<BasicBlock>(BB);
-            MDNode *md;
-            if (block->getFirstInsertionPt()->hasMetadataOtherThanDebugLoc())
-            {
-                md = block->getFirstInsertionPt()->getMetadata("BlockID");
-                if (md->getNumOperands() > 0)
-                {
-                    if (auto con = mdconst::dyn_extract<ConstantInt>(md->getOperand(0)))
-                    {
-                        int64_t blockID = con->getSExtValue();
-                        baseBlockMap[blockID] = block;
-                    }
-                }
-            }
+            F.setLinkage(GlobalValue::ExternalLinkage);
```

Comment:
Is this correct? I suspect this may break with code where we only link against some symbols, but not all. I'm thinking about where we still have to link to some so libraries.

Trying to start the conversation. Not sure what the answer is.

Reply:
I found that the linker didn't reliably work when this was absent. Somethings objects used in the Tik module would be defined in the original source bitcode. But (at least) a lot of the symbols in the original are closed to the outside world, can't remember what the keyword is. So I change them to the default external linkage so that they're allowed to link in another module